src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,65 @@
+package duke;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.util.Scanner;
+
+public class Duke {
+    private final String FILE_PATH = "data/duke.txt";
+    private Storage storage;
+    private TaskList inputList;
+    private Ui ui;
+
+    public Duke() {
+        ui = new Ui();
+        storage = new Storage(FILE_PATH);
+        try {
+            inputList = new TaskList(storage.load());
+        } catch (Exception e) {
+            System.out.println(e);
+            ui.showLoadingError();
+            inputList = new TaskList();
+        }
+    }
+
+    public void run() {
+        ui.identity();
+        Scanner userInput = new Scanner(System.in); // Create a Scanner object
+        String userOutput = userInput.nextLine(); // Read user input
+        while (!userOutput.equals("bye")) {
+            Boolean isBye = Parser.command(userOutput, inputList);
+            if (isBye) {
+                break;
+            }
```

Comment:
Not sure if the break is necessary since your while loop should end once the user types in "bye".